backend/app/views/spree/admin/store_credits/index.html.erb

```diff
@@ -70,6 +70,6 @@
   <div class="no-objects-found">
     <%= render 'spree/admin/shared/no_objects_found',
                  resource: Spree::StoreCredit,
-                 new_resource_url: new_object_url %>
+                 new_resource_url: spree.new_admin_user_store_credit_url %>
```

Comment:
I don't think it's bad to reference the URL helper directly, but it seems that the `new_object_url` shortcut is used extensively in the backend. Besides, it doesn't seem related to this PR. Can you help me understand why we need this change?

Reply:
it's the same issue than the one in users/index, it's a nested resource and it tries to build the url with two AR objects, so if the model that represents Spree.user_class is different than `user`, it does not match anymore with the routes.

`Spree.user_class = 'MyStore::Admin'` is going to generate new_admin_admin_store_credit_url, because it reads the model name, so, that's why we have to 'hardcode' the route name

Reply:
Thanks!